tests/specs/mcd/verification.k

```diff
@@ -109,4 +119,137 @@ module VERIFICATION
+    // (assert (forall ((x Int) (y Int)) (=> (and (= (sgn x) -1) (= (sgn y) -1)) (= (slt x y) (> x y)))))
+    //
+    // ;;;; proof
+    //
+    // ; uint256
+    // (declare-const x Int)
+    // (assert (and (>= x 0) (< x pow256)))
+    //
+    // (assert (not
+    //     (= (slt x 0) (= (sgn x) -1))
+    // ))
+    // (check-sat)
+    rule X s<Word 0 => bool2Word(sgn(X) ==Int -1) requires #rangeUInt(256, X)
+
+    rule sgn(chop(X)) ==Int -1 => X <Int 0 requires #rangeSInt(256, X)
+
+    // ### vat-mului-pass
+
+    rule chop(X *Int chop(Y)) => chop(X *Int Y)
+
+    // Proof #rangUInt(256, X) andBool #rangeSInt(256, Y) andBool #rangeSInt(256, X *Int Y) andBool Y =/=Int 0
+    //      chop(X *Int Y) /sWord chop(Y)
+    //   => #sgnInterp(sgn(chop(X *Int Y)) *Int sgn(chop(Y)), abs(chop(X *Int Y)) /Word abs(chop(Y)))
+    // Focus: sgn(chop(X *Int Y)) *Int sgn(chop(Y))
+    //      case 0 <=Int chop(Y)
+    //           sgn(chop(X *Int Y)) *Int sgn(chop(Y))
+    //        => sgn(chop(X *Int Y)) *Int sgn(Y)                     // chop(Y) ==Int Y
+    //        => sgn(X *Int Y) *Int sgn(Y)                           // 0 <=Int X andBool 0 <=Int Y
+    //        => 1 *Int 1                                            // 0 <=Int X *Int Y andBool 0 <=Int Y
+    //        => 1
+    //      case Y <Int 0
+    //           sgn(chop(X *Int Y)) *Int sgn(chop(Y))
+    //        => sgn(X *Int Y +Int pow256) *Int sgn(Y +Int pow256)   // chop(Y) ==Int Y +Int pow256
+    //        => -1 *Int -1                                          // X *Int Y <Int 0 andBool Y <Int 0
+    //        => 1
+    // Focus: abs(chop(X *Int Y)) /Word abs(chop(Y))
+    //      case 0 <=Int Y
+    //           abs(chop(X *Int Y)) /Word abs(chop(Y))
+    //        => abs(chop(X *Int Y)) /Int  abs(chop(Y))
+    //        => abs(X *Int Y) /Int abs(Y)                                                   // 0 <=Int X andBool 0 <=Int Y andBool #rangeSigne andBool chop(Y) ==Int Y
+    //        => X *Int Y /Int Y                                                             // #rangeSInt(256, X *Int Y) andBool 0 <=Int X *Int Y andBool #rangeSInt(256, Y) andBool 0 <=Int Y
+    //        => X
+    //      case Y <Int 0
+    //           abs(chop(X *Int Y)) /Int abs(chop(Y))
+    //        => abs(X *Int Y +Int pow256) /Int abs(Y +Int pow256)                           // #rangeSInt(256, X *Int Y) andBool chop(Y) ==Int Y +Int pow256
+    //        => (0 -Word (X *Int Y +Int pow256)) /Int (0 -Word (Y +Int pow256))             // #rangeSInt(256, X *Int Y) andBool X *Int Y <Int 0 andBool #rangeSInt(256, Y) andBool Y <Int 0
+    //        => (pow256 -Int (X *Int Y +Int pow256)) /Int (pow256 -Int (Y +Int pow256))     // 0 <Int (X *Int Y +Int pow256) andBool 0 <Int (Y +Int pow256)
+    //        => (X *Int Y) /Int Y
+    //        => X
+    //
+    // Via Z3 ?.?.?:
+    // ; 2^64 and 2^32 instead of 2^256 for small model proof
```

Comment:
They are 2^6 (= 64) and 2^5 (= 32). Quite small, but z3 takes too long beyond that.